feat(outlook): generate plugin xml manifest

chillfox opened the pull request.

stakach commented on src/controllers/outlook.cr:
```diff
@@ -0,0 +1,15 @@
+
+  get("/manifest.xml") do
+    manifest = OutlookManifest.new(
+      app_domain: "#{App::PLACE_URI}/outlook/",
```
we don't want to use `App::PLACE_URI` this is for internal comms between the containers
we need to use the current `tenant.domain` and only if `tenant.platform == "office365"`

chillfox replied:
Ok, can we get the tenant domain and platform without auth?
There is a requirement for the xml manifest to be served without auth.

chillfox replied:
I guess we could make the tenant part of the path, or a query param

stakach replied:
it doesn't need auth, just can't use the helper directly

```crystal
domain_host = request.hostname.as(String)
tenant = Tenant.query.find { domain == domain_host }
```

stakach reviewed (changes requested):
looks good, just one minor change

stakach reviewed (approved):
LGTM, let's hold off merging until the XML is finalised
@jeremyw24 to notify of any further changes required

stakach commented:
@jeremyw24 any additional XML changes now that the office plugin is working?

chillfox commented:
@stakach 
Since your last review of this the following changes has been made:
- updated xml
- added `outlook_config` json column to tenants for storing config.

stakach reviewed (approved):
Awesome work, looking good!
